### PR TITLE
Add Google Analytics for video views, track YouTube vs fallback, trac…

### PR DESCRIPTION
**Pull request by jeremydstone**
Track video plays in Google Analytics. Previously we only track video plays from Pegasus, not Dashboard, and didn't track if we were using YouTube or our fallback. Now we track Pegasus & Dashboard and YouTube vs fallback for both. We also track video downloads from Dashboard. (Deliberately not done: tracking video downloads from Pegasus, which is harder and appears low ROI to do.)

### CODE REVIEW

File: code-studio/src/js/videos.js [hunk 1]
```diff
@@ -228,17 +234,33 @@ function setupVideoFallback(videoInfo, playerWidth, playerHeight, shouldStillAdd
...
   }
 
-  videos.onYouTubeBlocked(function () {
-    if (!shouldStillAddCallback()) {
-      return;
-    }
-    addFallbackVideoPlayer(videoInfo, playerWidth, playerHeight);
-  });
+  videos.onYouTubeBlocked(
+      function () {
```
Comment (breville) [inline code-studio/src/js/videos.js]: 2 spaces per indentation, and i think `function() {` goes on same line as `onYouTubeBlocked`

style guide is at https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#javascript

File: code-studio/src/js/videos.js [hunk 2]
```diff
@@ -228,17 +234,33 @@ function setupVideoFallback(videoInfo, playerWidth, playerHeight, shouldStillAdd
...
+      videoInfo);
 }
 
 // This is exported (and placed on window) because it gets accessed externally for our video test page.
-videos.onYouTubeBlocked = function (callback) {
-  testImageAccess(youTubeAvailabilityEndpointURL() + '?' + Math.random(), function (){}, callback);
+videos.onYouTubeBlocked = function (youTubeBlockedCallback, videoInfo) {
+  var key = (videoInfo ? videoInfo.key : undefined);
+  testImageAccess(youTubeAvailabilityEndpointURL() + '?' + Math.random(),
+      // Called when YouTube availability check succeeds.
```
Comment (breville) [inline code-studio/src/js/videos.js:249]: nice commenting and expansion.

File: pegasus/sites.v3/code.org/views/display_video_lightbox.haml [hunk 3]
```diff
@@ -27,7 +27,13 @@
     setupPopupWindows();
     window["playVideo_#{id}"](true);
 
-    ga('send', 'event', 'showvideo', 'click', {'id': "#{id}"});
+    // send video start action to Google Analytics
```
Comment (breville) [inline pegasus/sites.v3/code.org/views/display_video_lightbox.haml]: orphan comment?  combine with below comment?

File: pegasus/sites.v3/code.org/views/display_video_lightbox.haml [hunk 4]
```diff
@@ -27,7 +27,13 @@
...
+
+    // change action tag to fallback if using fallback video player serving from Code.org CLoudFront
```
Comment (breville) [inline pegasus/sites.v3/code.org/views/display_video_lightbox.haml]: nit: CloudFront.

**Comment by breville** [comment]
lgtm.  looking forward to seeing this data.

**Comment by jeremydstone** [comment]
Thanks. Updated per your comments.

**Pull request merged**